src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -160,15 +180,19 @@ public static int Log2(ulong value)
 
             if (Lzcnt.X64.IsSupported)
             {
-                // LZCNT contract is 0->64
-                return 63 - (int)Lzcnt.X64.LeadingZeroCount(value);
+                return 63 ^ (int)Lzcnt.X64.LeadingZeroCount(value);
             }
 
             if (ArmBase.Arm64.IsSupported)
             {
                 return 63 - ArmBase.Arm64.LeadingZeroCount(value);
             }
 
+            if (X86Base.IsSupported)
+            {
+                return (int)X86Base.X64.BitScanReverse(value);
```

Comment:
x86 test failures come from the incorrect `IsSupported` check here.  JIT side appears to be fine